components/ToDoContainer.vue

```diff
@@ -1,21 +1,78 @@
 <script setup lang="ts">
+import mockTasks from '@/mocks/tasks.js';
 //business component with logic (storage, filtering, adding, deleting)
 
+const tasks = ref(mockTasks);
 const searchBy = ref('');
+const removeTaskModal = ref(false);
+const addTaskModal = ref(false);
+const taskIdToRemove = ref();
 
-const handleSearch = (value) => {
-  console.log(value);
+const getFormattedDate = (): string => {
+  const now = new Date();
+  const day = now.getDate();
+  const month = now.toLocaleString('en-US', { month: 'long' });
 
+  return `${day} ${month}`;
+};
+
+const addTask = (value: string) => {
+  tasks.value.push({
+    title: value,
+    isDone: false,
+    date: getFormattedDate(),
+    id: Math.random(),
+  });
+
+  addTaskModal.value = false;
+};
+
+const handleSearch = (value: string) => {
   searchBy.value = value;
 };
 
-const handleDelete = (id: string) => {
-  tasks.value = tasks.value.filter(task => task.id !== id);
+const filteredTasks = computed(() => {
+  if (!searchBy.value) return tasks.value;
+
+  return tasks.value.filter(task =>
+    task.title.toLowerCase().includes(searchBy.value.toLowerCase())
+  );
+});
+
+const completedTasks = computed(() => {
+  return tasks.value.filter(task => task.isDone).length;
+});
+
+const handleDeleteTask = (taskId: string) => {
+  removeTaskModal.value = true;
+
+  taskIdToRemove.value = taskId;
 };
 
-const handleComplete = (id: string) => {
+const acceptDeleting = () => {
+  tasks.value = tasks.value.filter(task => task.id !== taskIdToRemove.value);
+
+  removeTaskModal.value = false;
+}
+
+const rejectDeleting = ()=>{
+  taskIdToRemove.value = '';
 
+  removeTaskModal.value = false;
+}
+
+const handleAddTask = () => {
```

Comment:
It's unclear what action is fired by this method. If it's opening a modal, you have to provide more details to let people understand it at first read.
**_openAddTaskModal_**